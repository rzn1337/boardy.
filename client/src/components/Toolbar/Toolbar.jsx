import { PencilLine, Minus, RectangleHorizontal, Move, Scaling } from "lucide-react";

function Toolbar({ setTool }) {
    const tools = [
        { name: "Select", icon: <Move strokeWidth={2}/> },
        { name: "Rectangle", icon: <RectangleHorizontal strokeWidth={2}/> },
        { name: "Resize", icon: <Scaling strokeWidth={2}/> },
        // { name: "Arrow", icon: "â†’" },
        { name: "Line", icon: <Minus strokeWidth={2}/> },
        { name: "Freedraw", icon: <PencilLine strokeWidth={2}/> },
        
        // { name: "Text", icon: "T" },
        // { name: "Image", icon: "ðŸ–¼" },
        // { name: "Eraser", icon: "âŒ«" },
    ];
    return (
        <div className="fixed top-4 left-1/2 transform -translate-x-1/2 bg-slate-900 rounded-xl shadow-md">
            <div className="flex p-1">
                {tools.map((tool) => (
                    <button
                        key={tool.name}
                        onClick={() => setTool(tool.name.toLowerCase())}
                        className={`w-9 h-9 flex items-center justify-center rounded-md mx-0.5 text-white hover:bg-gray-100 `}
                        title={tool.name}
                    >
                        {tool.icon}
                    </button>
                ))}
            </div>
        </div>
    );
}

export default Toolbar;
