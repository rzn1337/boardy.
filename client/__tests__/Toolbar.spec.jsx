import { describe, test, expect } from "vitest";
import { render } from "@testing-library/react";
import Toolbar from "../src/components/Toolbar/Toolbar";

describe("Canvas", () => {
    
    test("should render toolbar", () => {
        const result = render(<Toolbar />);
        expect(result.container).toMatchSnapshot();
    });
});
