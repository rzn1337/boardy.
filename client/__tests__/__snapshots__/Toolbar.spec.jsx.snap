// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Canvas > should render toolbar 1`] = `
<div>
  <div
    class="fixed top-4 left-1/2 transform -translate-x-1/2 bg-slate-900 rounded-xl shadow-md"
  >
    <div
      class="flex p-1"
    >
      <button
        class="w-9 h-9 flex items-center justify-center rounded-md mx-0.5 text-white hover:bg-gray-100 "
        title="Select"
      >
        <svg
          class="lucide lucide-move"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="5 9 2 12 5 15"
          />
          <polyline
            points="9 5 12 2 15 5"
          />
          <polyline
            points="15 19 12 22 9 19"
          />
          <polyline
            points="19 9 22 12 19 15"
          />
          <line
            x1="2"
            x2="22"
            y1="12"
            y2="12"
          />
          <line
            x1="12"
            x2="12"
            y1="2"
            y2="22"
          />
        </svg>
      </button>
      <button
        class="w-9 h-9 flex items-center justify-center rounded-md mx-0.5 text-white hover:bg-gray-100 "
        title="Rectangle"
      >
        <svg
          class="lucide lucide-rectangle-horizontal"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="12"
            rx="2"
            width="20"
            x="2"
            y="6"
          />
        </svg>
      </button>
      <button
        class="w-9 h-9 flex items-center justify-center rounded-md mx-0.5 text-white hover:bg-gray-100 "
        title="Resize"
      >
        <svg
          class="lucide lucide-scaling"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
          />
          <path
            d="M14 15H9v-5"
          />
          <path
            d="M16 3h5v5"
          />
          <path
            d="M21 3 9 15"
          />
        </svg>
      </button>
      <button
        class="w-9 h-9 flex items-center justify-center rounded-md mx-0.5 text-white hover:bg-gray-100 "
        title="Line"
      >
        <svg
          class="lucide lucide-minus"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 12h14"
          />
        </svg>
      </button>
      <button
        class="w-9 h-9 flex items-center justify-center rounded-md mx-0.5 text-white hover:bg-gray-100 "
        title="Freedraw"
      >
        <svg
          class="lucide lucide-pencil-line"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 20h9"
          />
          <path
            d="M16.376 3.622a1 1 0 0 1 3.002 3.002L7.368 18.635a2 2 0 0 1-.855.506l-2.872.838a.5.5 0 0 1-.62-.62l.838-2.872a2 2 0 0 1 .506-.854z"
          />
          <path
            d="m15 5 3 3"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
